xbmc/favourites/GUIWindowFavourites.cpp

```diff
@@ -111,20 +111,24 @@ class CVideoPlayActionProcessor : public VIDEO::GUILIB::CVideoPlayActionProcesso
 };
 } // namespace
 
-bool CGUIWindowFavourites::OnSelect(int item)
+bool CGUIWindowFavourites::OnSelect(int itemIdx)
 {
-  if (item < 0 || item >= m_vecItems->Size())
+  if (itemIdx < 0 || itemIdx >= m_vecItems->Size())
     return false;
 
-  const CFavouritesURL favURL{*(*m_vecItems)[item], GetID()};
+  const auto item{(*m_vecItems)[itemIdx]};
+  const CFavouritesURL favURL{*item, GetID()};
   if (!favURL.IsValid())
     return false;
 
-  CFileItem targetItem{favURL.GetTarget(), favURL.IsDir()};
-  targetItem.LoadDetails();
-
   const bool isPlayMedia{favURL.GetAction() == CFavouritesURL::Action::PLAY_MEDIA};
 
+  const auto target{CServiceBroker::GetFavouritesService().ResolveFavourite(*item)};
+  if (!target)
+    return false;
+
+  CFileItem targetItem{*target};
```

Comment:
Is the copy needed?

Reply:
Maybe not, was aking myself the same, then simply took over the pattern from existing code. Will put to check this on my todo list for later.